### PR TITLE
Add possibility to leave from workspaces and workspace expense chats

### PR DESCRIPTION
**Pull request by rezkiy37**
### Details

The PR introduces a couple flows for leaving a workspace and a workspace chat. It is based on rules that described [here](https://github.com/Expensify/App/pull/34987/files#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5) for these scenarios.

### Fixed Issues

$ https://github.com/Expensify/App/issues/35286
PROPOSAL: N/A

#### Leaving a workspace

1. A current user shouldn't be a creator or an admin of a workspace you want to leave.
1.1. So invite the user from another account to a workspace. You can do it via the workspace settings members page.
3. Open the workspaces page.
4. Hover on a "three dots" button on a workspace.
5. Verify that there is a button "Leave" for a workspace that satisfies the 1st point.
6. Click on the button.
7. Verify that the workspace has gone from the list.
8. Verify that the #announce room has been archived.
9. Verify other related chats have been archived.

#### Leaving a workspace chat

1. Login with user A
2. Create or join a workspace if not done previously.
3. In user A expense chat, mention user B and click invite them on the whisper message (_user B shouldn't be an admin or a direct member (default) of the workspace chat you want to leave_).
3. Login with user B
4. Open the workspace chat page.
5. Hover on a "three dots" button on a header.
6. Verify that there is a button "Leave" for a workspace that satisfies the 1st point.
7. Click on the button.
8. Verify that the workspace chat has gone from the LHN.

- [x] Verify that no errors appear in the JS console

### Offline tests

Same as **"Tests"**.

### QA Steps

Same as **"Tests"**.

- [x] Verify that no errors appear in the JS console

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>

https://github.com/Expensify/App/assets/57314631/dad39303-7758-4007-885e-33e9fc216119

</details>

<details>
<summary>Android: mWeb Chrome</summary>

https://github.com/Expensify/App/assets/57314631/1a521aeb-15ac-49ab-8352-d1dbc735b999

</details>

<details>
<summary>iOS: Native</summary>

https://github.com/Expensify/App/assets/57314631/02e6a455-4fe2-4a41-a006-0cbe5aafb0aa

</details>

<details>
<summary>iOS: mWeb Safari</summary>

https://github.com/Expensify/App/assets/57314631/75ad4278-b1e8-4e6b-aeda-b5ac15294822

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

https://github.com/Expensify/App/assets/57314631/be325c82-f65a-49a5-9bb8-f0c626f0e2b6

https://github.com/Expensify/App/assets/57314631/a1f8e36e-fb6f-4911-b852-75624736ea39

</details>

<details>
<summary>MacOS: Desktop</summary>

https://github.com/Expensify/App/assets/57314631/0b016b24-24d0-4afe-8767-bffd0b24f78d

</details>

### CODE REVIEW

**Comment by MonilBhavsar** [comment]
@rezkiy37 can we please resolve this conflicts 🙏

**Comment by rezkiy37** [comment]
@MonilBhavsar, you can try to use this PR for the testing.

File: src/pages/workspace/WorkspacesListPage.tsx
Comment (MonilBhavsar) [inline src/pages/workspace/WorkspacesListPage.tsx:177]: @rezkiy37 we're now using a new command for Leaving workspace case - 
`LeavePolicy`
that accepts parameters - authToken: string, policyID: string, email: string

Can you please update that. The API is not live on staging yet. Should be live today or latest by tomorrow.
Thanks!

Comment (rezkiy37) [reply src/pages/workspace/WorkspacesListPage.tsx:177]: Integrated a draft implementation - ea03623.

Comment (MonilBhavsar) [reply src/pages/workspace/WorkspacesListPage.tsx:177]: Thanks! Looks good to me. I'll notify once it is ready to be tested!
We also need to define a changeLog action - `'POLICYCHANGELOG_LEAVE_POLICY'` with the message - `left the workspace`

Comment (rezkiy37) [reply src/pages/workspace/WorkspacesListPage.tsx:177]: I added a const - 5ca68a6. However, I don't see any examples of integration message sending and log tracking. Can you help me here?

Comment (MonilBhavsar) [reply src/pages/workspace/WorkspacesListPage.tsx:177]: When a workspace member leaves the workspace, the #admins room of the corresponding workspace should have a `reportAction` with action - `POLICYCHANGELOG_LEAVE_POLICY`. You can view it from the admin account of that workspace

Comment (rezkiy37) [reply src/pages/workspace/WorkspacesListPage.tsx:177]: Sorry, I just realized that admins cannot leave workspaces. There is an expected restriction for admins and owners of workspaces. So, this report action is generated on the backend side and shared with other admins in `#admins` rooms. More info [here](https://github.com/Expensify/App?tab=readme-ov-file#workspace-1).


Screenshot 2024-03-27 at 15 11 37

Comment (MonilBhavsar) [reply src/pages/workspace/WorkspacesListPage.tsx:177]: Yes, that's all correct

**Comment by rezkiy37** [comment]
Update - https://github.com/Expensify/App/issues/35286#issuecomment-2027201045.

**Review by rayane-d — CHANGES_REQUESTED** [review]
Code looks good! I've left a few minor comments.

**Comment by rezkiy37** [comment]
> Bug1:
> 
>  Recording.2024-04-04.024500.1.mp4 
> Steps:
> 
> 1. With User A Invite User B to a workspace
> 2. Login with user B
> 3. Send a few messages, create tasks and IOU actions
> 4. Turn on offline mode
> 5. Leave the workspace
> 
> Expected Result: User B is not able to add comments and actions in reports (all reports should be archived)
> 
> Actual result: Only the expense report is archived and user B can perform actions in other reports. when going back online, red dots are displayed with API errors

I confirm that the bug existed. The problem was that `getWorkspaceChats` gets only a policy expense chat, where the user is an owner. I've created a new function `getAllWorkspaceReports` to get all associated reports to a workspace. Done - 9e9f7bf.

cc @rayane-djouah 

https://github.com/Expensify/App/assets/57314631/4b2b1d79-dc4c-4130-9072-8e4ea3586e21

**Comment by MonilBhavsar** [comment]
Can we also check the offline behavior please. It should be similar to removing workspace members

**Comment by rezkiy37** [comment]
> Can we also check the offline behavior, please. It should be similar to removing workspace members

[Here](https://github.com/Expensify/App/pull/35671#issuecomment-2037148915) you can see how it works offline, and then I restore the connection. Also, please read my comment. It works slightly differently, because as was [mentioned](https://github.com/Expensify/App/pull/35671#issuecomment-2035982107) before, the app needs to archive all related chats.

**Comment by rezkiy37** [comment]
> > > To make test steps clear, please add in the "Leaving a workspace chat" section how to add a user to a workspace chat that doesn't belong to them (I believe the only option available is by mention?)
> > 
> > 
> > Done.
> 
> I meant Test 2 (Leaving a workspace chat); I think that these steps are more apparent:
> 
> 1. Login with user A
> 2. Create or join a workspace if not done previously.
> 3. In user A expense chat, mention user B and click invite them on the whisper message (user B shouldn't be an admin or a direct member (default) of the workspace chat you want to leave).
> 4. Login with user B
> 5. Open the workspace chat page.
> 6. Hover on a "three dots" button on a header.
> 7. Verify that there is a button "Leave" for a workspace that satisfies the 1st point.
> 8. Click on the button.
> 9. Verify that the workspace chat has gone from the LHN.

Ah, sorry, My bad. I will update.

**Comment by rezkiy37** [comment]
> Prettier check is failing

Fixed.

**Comment by rezkiy37** [comment]
> @rezkiy37 Typescript check is failing, `hasDraft` was replaced in #37281

Fixed.

**Comment by rezkiy37** [comment]
> Bug 2:
> 
> Steps:
> 
> 1. From user A, invite user B to a workspace expense chat by mention
> 2. From user B, open the report and copy its link
> 3. From user B, leave the report
> 4. From user B, visit the report link
> 
> Expected result: User B should see a join button (the same behaviour as rooms/threads), but no leave option is provided.
> 
> Actual result: No join button is displayed, and the leave option still exists.
> 
>  Recording.2024-04-05.162307.mp4 
> cc @rezkiy37

I can confirm that the bug exists. However, it looks like the bug exists on the backend side. When the app is offline, it restricts access to the room.

Btw, what does `reportUserIsLeavingRoom_` mean? Can we use it for this case?

cc @MonilBhavsar 

https://github.com/Expensify/App/assets/57314631/b66cc6d0-c845-485f-9481-ebc4ba82abc2

Screenshot 2024-04-05 at 18 40 09

**Comment by MonilBhavsar** [comment]
Can anyone please help to understand how bug #2 is related to Leave Workspace flow. User is not leaving a workspace at all?

**Comment by rezkiy37** [comment]
I've investigated the case with bug #2. 

1. There is an important difference [if the user is a workspace member](https://github.com/Expensify/App/blob/dd30d137b485482bb8e4c61700b58d2e03683124/src/pages/home/HeaderView.tsx#L155) when the user is leaving a room (_not an expense report_). I mean the user [can rejoin](https://github.com/Expensify/App/blob/87747ad36fe14cfd887cd72f249a93f7c1670530/src/libs/actions/Report.ts#L2364-L2366) the room only if the user is a workspace member. The backend responses also follow this scenario. When the user is not a workspace member, the user cannot rejoin the room. The app shows a loading view. Looks like it is the correct behavior. WDYT, @MonilBhavsar?

<details><summary>Videos:</summary>

https://github.com/Expensify/App/assets/57314631/23e3baca-3b46-41d6-b6ff-052b414a68f4


https://github.com/Expensify/App/assets/57314631/84853288-bb7f-4bdc-bbe9-35f0f6f51ac5

</details> 

2. Regarding the case with expense chats, I fixed (2c2ae9f) a condition to check if the user is leaving an expense report. It works pretty the same as for regular rooms when the user is a member of a workspace. However, I've found a strange behaviour that the backend keeps access for the expense report once the user has left it when the user is not a member of the workspace. AFAIK, the user shouldn't be able to open this expense report.

<details><summary>Videos:</summary>


https://github.com/Expensify/App/assets/57314631/df3cd5ca-7025-4e29-9b96-ca40a38ba0dd


https://github.com/Expensify/App/assets/57314631/b723273d-93d2-478e-b6cf-a01adfb6fc33


</details> 

cc @rayane-djouah

**Review by rayane-d — CHANGES_REQUESTED** [review]
Please check above comments

**Comment by MonilBhavsar** [comment]
> To summarise the current behaviour of the ability to rejoin a report after leaving it:

Yeah, that sounds bug from backend then. Only a workspace member should be able to join policy expense chat. I'll take a look

**Comment by rezkiy37** [comment]
> Bug 3:
> 
> Steps:
> 
> 1. With User A, Invite User B to a workspace
> 2. Login with user B
> 3. Leave the workspace
>    Repeat the steps a few times.
> 
> Expected Result: User A should see one expense chat report for user B
> 
> Actual result: User A see multiple expense chat report for user B, one is valid and others have not found error.
> 
> image
> 
> image

The problem was because [getChatByParticipantsAndPolicy](https://github.com/Expensify/App/blob/4f673576491af54296014b7595e6ad5017fdb60e/src/libs/ReportUtils.ts#L4578) function compares strictly. However, sometimes there is one more participant. It is an Exfy support person. So, I've fixed (858a872) this condition to check if a legacy report includes current IDs.

<details><summary>Details</summary>
<p>


https://github.com/Expensify/App/assets/57314631/9e92970f-432a-4087-b8f6-2ec54b3723cd

Screenshot 2024-04-09 at 12 54 31


</p>
</details>

**Comment by rezkiy37** [comment]
> Bug 4:
> 
> Steps:
> 
> 1. With User A, Invite User B to a workspace
> 2. Login with user B
> 3. Leave the workspace
> 4. With user A, re-invite user B to the workspace
> 
> Expected Result: User B should see its expense chat report subtitle in LHN as it was before leaving or the invited message from the whisper.
> 
> Actual result:
> 
> User B see `closed report` in the LHN subtitle of its expense chat report.
> 
>  Recording.2024-04-09.005227.mp4 
> and user A sees:
> 
> image

I've investigated the bug. Looks like it is a backend issue. There is no pusher message to update text in **LHN**. Once user A, re-invite user B to the workspace, the backend should send updates for both users. cc @MonilBhavsar 

Screenshot 2024-04-09 at 13 01 55

**Comment by rezkiy37** [comment]
> Bug 5:
> 
> Steps:
> 
> 1. With User A, Invite User B to a workspace
> 2. With User A, Invite User B to User A's expense chat report by mention.
> 3. Login with user B
> 4. Leave the workspace
> 
> Expected Result: User B should see User A's expense chat report with no changes (not archived).
> 
> Actual result: User B see User A's expense chat report as archived for a moment before it goes back as not archived.
> 
>  Recording.2024-04-09.010109.mp4

I've found the root cause of the bug. I added (0b12883) a couple conditions to check if the user should leave an expense report when the user is not an owner.

<details><summary>Video</summary>
<p>

https://github.com/Expensify/App/assets/57314631/cc2f1f64-b02b-4393-99a3-b63109cc00d3

</p>
</details>

**Comment by rezkiy37** [comment]
> Bug 6: "Hmm... it's not here" page when opening the announce room after rejoining the workspace after leaving. Not sure about the test steps.
> 
> image

I've investigated the bug. It happens because the app sets the "not found" error to the report. It blocks the showing of the page even when the report exists. So, I added a condition (4de342a) to check if the report should have the error, if not the app clears.

<details><summary>Video</summary>
<p>

https://github.com/Expensify/App/assets/57314631/2a14cf1d-2c13-4e17-a29a-6b9345bbfc3e

</p>
</details>

**Comment by rezkiy37** [comment]
@rayane-djouah, I've resolved all bugs. Please review it one more time. Also, we are waiting on the backend fixes.

**Comment by MonilBhavsar** [comment]
It is sent, may be we're not displaying it?
https://github.com/Expensify/App/pull/35671#discussion_r1539096336

**Review by rayane-d — CHANGES_REQUESTED** [review]
Please fix bug [6](https://github.com/Expensify/App/pull/35671#issuecomment-2045481675) and [7](https://github.com/Expensify/App/pull/35671#issuecomment-2045441934)

**Comment by rezkiy37** [comment]
Also, updated the sidebar message - 8ab074e.

<details><summary>Screenshot</summary>
<p>

Screenshot 2024-04-10 at 14 24 11

</p>
</details>

**Comment by MonilBhavsar** [comment]
> Do we need to hold for backend fix for these: https://github.com/Expensify/App/pull/35671#issuecomment-2043551587, https://github.com/Expensify/App/pull/35671#issuecomment-2044730156?

Let's not hold on to that 👍

File: src/languages/es.ts
Comment (rezkiy37) [reply src/languages/es.ts:2175]: [Asked](https://expensify.slack.com/archives/C01GTK53T8Q/p1712828491926959).

File: src/libs/actions/Policy.ts
Comment (MonilBhavsar) [inline src/libs/actions/Policy.ts:1067]: Do we need to explicitly pass authToken param?

Comment (rezkiy37) [reply src/libs/actions/Policy.ts]: Good catch! It is a redundant property to be passed explicitly. Omitted - 97cab94.

<details><summary>Video</summary>
<p>


https://github.com/Expensify/App/assets/57314631/0124ffff-2d17-4475-b78f-35f0078ac995



</p>
</details>

**Comment by rezkiy37** [comment]
> @rezkiy37 [0b12883](https://github.com/Expensify/App/commit/0b12883e71655d71e2dd545c92842b0ea54c2601) didn't handle the case of threads (`reportOwnerID` is 0):
> 
> image

I've added a condition to check a parent report - 5313900.

<details><summary>Video</summary>
<p>

https://github.com/Expensify/App/assets/57314631/370e7bd3-ddbe-4db1-ad5f-c0ebc6f285b7

</p>
</details>

**Comment by rezkiy37** [comment]
> Bug: endless LeavePolicy requests after logout and logging back:
> 
>  Recording.2024-04-10.231629.mp4

Sorry, I need more context on how to reproduce the bug. Also, I don't see any specific configuration for this request. Is it a common bug across the app?

**Comment by MonilBhavsar** [comment]
There are some conflicts to resolve and @rayane-djouah looks like last bug is not reproducible. Can you please take a look?

**Review by rayane-d — APPROVED** [review]
LGTM and tests well 🚀

File: src/libs/ReportActionsUtils.ts
Comment (MonilBhavsar) [inline src/libs/ReportActionsUtils.ts:873]:
```suggestion
```

File: src/libs/ReportActionsUtils.ts
Comment (MonilBhavsar) [inline src/libs/ReportActionsUtils.ts:173]:
```suggestion
function isLeavePolicyAction(reportAction: OnyxEntry<ReportAction>) {
```

File: src/libs/actions/Policy.ts
Comment (MonilBhavsar) [inline src/libs/actions/Policy.ts:977]: We can remove this. name implies the reason
```suggestion
```

**Review by MonilBhavsar — COMMENTED** [review]
Thanks! Looks good overall

File: src/libs/ReportActionsUtils.ts
Comment (rezkiy37) [reply src/libs/ReportActionsUtils.ts]: Done - 7321ad2.

File: src/libs/ReportActionsUtils.ts
Comment (rezkiy37) [reply src/libs/ReportActionsUtils.ts]: Done - 7321ad2.

File: src/libs/actions/Policy.ts
Comment (rezkiy37) [reply src/libs/actions/Policy.ts]: Done - 7321ad2.

**Review by MonilBhavsar — APPROVED** [review]
Thank you! Let's ship it

**Comment by MonilBhavsar** [comment]
@rayane-djouah @rezkiy37 there were just this two backend bugs to fix https://github.com/Expensify/App/pull/35671#issuecomment-2048388145, right?

**Comment by MonilBhavsar** [comment]
Thank you for confirming!

**Comment by MonilBhavsar** [comment]
@getusha @rezkiy37 this is the exact issue we discussed here, right?
https://github.com/Expensify/App/pull/35671#issuecomment-2043551587
https://github.com/Expensify/App/issues/40417

**Comment by MonilBhavsar** [comment]
and as per the security guidelines, a non workspace member can join the workspace chat https://github.com/Expensify/App?tab=readme-ov-file#security. So I think we should update both frontend and backend to allow anyone to rejoin the chat. What do you think?

**Comment by rezkiy37** [comment]
> and as per the security guidelines, a non workspace member can join the workspace chat https://github.com/Expensify/App?tab=readme-ov-file#security. So I think we should update both frontend and backend to allow anyone to rejoin the chat. What do you think?

Can we create an issue for this or do you want to fix in a scope of current issue?

**Comment by rezkiy37** [comment]
Hey!
I just wanted to let you know about my vacation for next week. I will be back on 06.05.2024. Meanwhile, I asked @waterim to take over this issue. 
See you soon!

File: src/pages/home/ReportScreen.tsx
Comment (arosiclair) [inline src/pages/home/ReportScreen.tsx:400]: This doesn't seem right to me. Why is this needed? @rezkiy37

Comment (rezkiy37) [reply src/pages/home/ReportScreen.tsx:400]: @arosiclair, please check the condition above. The app checks if a report has `reportID`, if yes it means that the `notFound` error is not relevant anymore.

Comment (arosiclair) [reply src/pages/home/ReportScreen.tsx:400]: That's not necessarily true. If you lose access to a report,`report.errorFields.notFound` will be set along with the rest of the report data including the `reportID`.

So why is this needed? `errorFields.notFound` should already be cleared in the [successData for OpenReport](https://github.com/Expensify/App/blob/main/src/libs/actions/Report.ts#L786-L794).

Comment (rezkiy37) [reply src/pages/home/ReportScreen.tsx:400]: There was a bug - https://github.com/Expensify/App/pull/35671#issuecomment-2045481675. I fixed it - https://github.com/Expensify/App/pull/35671#issuecomment-2045352094.

Comment (arosiclair) [reply src/pages/home/ReportScreen.tsx:400]: Okay it sounds like the `notFound` error is not cleared when we push reports to users. I think we can just fix that on the backend. Do you know which API command you were testing that caused that bug? cc @MonilBhavsar

Comment (rezkiy37) [reply src/pages/home/ReportScreen.tsx:400]: Looks like **InviteToRoom**

Comment (arosiclair) [reply src/pages/home/ReportScreen.tsx:400]: Alright looks like we just need to update [Report::structureReportForOnyx](https://github.com/Expensify/Auth/blob/5110e614fe472804988dbb845536a3565e138016/auth/lib/Report.cpp#L8632) to clear `errorFields.notFound`. I opened an issue to clean this up [here](https://github.com/Expensify/App/issues/43357).

File: src/pages/workspace/WorkspacesListPage.tsx
Comment (mollfpr) [inline src/pages/workspace/WorkspacesListPage.tsx:175]: Hi 👋 coming from https://github.com/Expensify/App/issues/46248.

Pretty minor issue and part of the polish issue. We should be using the `Exit` icon.

File: src/libs/actions/Policy.ts
Comment (ahmedGaber93) [inline src/libs/actions/Policy.ts:1025]: Coming from here https://github.com/Expensify/App/issues/56516, we fallback the `pendingAction` to null if `policy?.pendingAction` is undefined to re-display the workspace item if leaveWorkspace API request is failed

File: src/pages/workspace/WorkspacesListPage.tsx
Comment (dukenv0307) [inline src/pages/workspace/WorkspacesListPage.tsx:173]: Coming from https://github.com/Expensify/App/issues/68355, we don't want to remove the default approval

Comment (situchan) [reply src/pages/workspace/WorkspacesListPage.tsx:173]: Coming from https://github.com/Expensify/App/issues/71661, member added to group with preferred workspace should not have option to leave workspace.

File: src/libs/actions/Policy.ts
Comment (rojiphil) [inline src/libs/actions/Policy.ts:1037]: A special case got missed where the employee is made as the approver and the employee leaves the workspace. Here, the owner workspace chat should not be displayed in LHN which we fixed in #75329

**Pull request merged**
